src/test/java/dataloader/JsonLoaderTest.java

```diff
@@ -0,0 +1,74 @@
+package dataloader;
+
+import org.junit.Test;
+import tech.tablesaw.api.Table;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+
+import static junit.framework.Assert.*;
+
+public class JsonLoaderTest {
+
+    /**
+     * Test Scenario: The application wants to read a json dataset
+     * but the String is invalid(empty String).
+     * Expected Output: JsonDataLoader throws an IOException.
+     */
+
+    @Test(expected = IOException.class)
+    public void testEmptyStringForHashMap() throws IOException {
+        JsonDataLoader loader = new JsonDataLoader();
+        loader.parseFile("nonExistent.json", new HashMap<String, Object>());
+    }
+
+    /**
+     * This is the same test as above
+     * but for type Table.
+     */
+
+    @Test(expected = IOException.class)
+    public void testEmptyStringForTable() throws IOException {
+        JsonDataLoader loader = new JsonDataLoader();
+        Table dataSet = null;
+        loader.parseFile("nonExistent.json", dataSet);
+    }
+
+    /**
+     * Test Scenario: a json file is provided
+     * Expected Output: the dataset is returned
+     * along with the correct column size and row count.
+     */
+
+    @Test
+    public void testJsonFileForTable() throws IOException {
+        JsonDataLoader loader = new JsonDataLoader();
+        Table dataSet = null;
+        File file = new File("test_data/dummy_array.json");
+        dataSet = loader.parseFile(file, dataSet);
+
+        assertNotNull(dataSet);
+        assertEquals(dataSet.columns().size(), 3);
+        assertEquals(dataSet.rowCount(), 1);
+    }
+
+    /**
+     * Test Scenario: The user provides a json file that is not
+     * in array format.
+     * Expected Output: The application read the json file and
+     * return a data set with the correct key-value pairs.
+     * */
+    @Test
+    public void testJsonFileForHashMap() throws IOException {
+        JsonDataLoader loader = new JsonDataLoader();
+        HashMap dataSet = new HashMap<String, Object>();
+        loader.parseFile("test_data/dummy.json", dataSet);
+
+        assertNotNull(dataSet);
```

Comment:
This will never fail the way ```parseFile``` is implemented. Perhaps is better that we shift to returning the data set object and not passing it as an argument

Reply:
will check it again. tdd....